### PR TITLE
chore: lender contracts

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
main
- merged reserve with operator
- merged slicer with operator
- removed trancheManager dependency from assessor
- senior tranche uses pie concept

minor
- update debt splitted in senior operator

### CODE REVIEW

File: src/core/tranche/senior_tranche.sol [hunk 1]
```diff
@@ -15,73 +15,63 @@
...
+        super.borrow(usr, currencyAmount);
     }
 
     function drip() internal {
-        if (now >= borrowRate.lastUpdated) {
-            (uint latest, , uint wad) = compounding();
-            borrowRate.index = latest;
-            borrowRate.lastUpdated = uint48(now);
-            debt = add(debt, wad);   
+        if (now >= lastUpdated) {
```
Comment (charleenfei) [inline src/core/tranche/senior_tranche.sol]: 🏅

File: src/core/tranche/tranche.sol [hunk 2]
```diff
@@ -40,40 +44,58 @@ contract Reserve is DSNote {
...
+
+        self = address(this);
     }
 
     function balance() public returns (uint) {
-        return currency.balanceOf(address(this));
+        return currency.balanceOf(self);
     }
 
     function tokenSupply() public returns (uint) {
```
Comment (charleenfei) [inline src/core/tranche/tranche.sol]: idk if this function is necessary, we could just used `token.totalSupply()` directly

File: src/core/tranche/assessor.sol [hunk 3]
```diff
@@ -16,17 +16,21 @@
...
+contract SeniorTrancheLike {
     function debt() public returns(uint);
 }
 
 contract PileLike {
     function Debt() public returns(uint);
 }
 
-contract Assessor is DSNote {
+contract Assessor is DSNote,DSMath {
```
Comment (ilinzweilin) [inline src/core/tranche/assessor.sol]: `contract Assessor is DSNote, DSMath {`

File: src/core/tranche/assessor.sol [hunk 4]
```diff
@@ -47,30 +51,26 @@ contract Assessor is DSNote {
...
     }
 
-    function getAssetValue() public returns (uint) {
-         OperatorLike operator = OperatorLike(msg.sender);
-         uint trancheReserve = operator.balance();
-         uint poolValue = pile.Debt();
-         if (msg.sender == junior) {
-            return calcJuniorAssetValue(poolValue, trancheReserve, calcSeniorDebt());
-         }
+    function calcTokenPrice() public returns (uint) {
```
Comment (ilinzweilin) [inline src/core/tranche/assessor.sol]: maybe removing calc & just naming tokenPrice. same for calcJuniorAssetValue & calcSeniorAssetValue

**Comment by manuelpolzhofer** [comment]
Decided to merge the PR together with Charly and Alina. Refactoring of the tests will happen in the next PR's

**Pull request merged**